libvips/colour/icc_transform.c

```diff
@@ -617,6 +618,9 @@ vips_icc_load_profile_blob(VipsBlob *blob,
 		return NULL;
 	}
 
+	if (icc->intent == VIPS_INTENT_AUTO)
+		icc->intent = (VipsIntent) cmsGetHeaderRenderingIntent(profile);
```

Comment:
Will this break caching? Perhaps have a `VipsIntent selected_intent;` member that's not a constructor parameter.